pkg/eden/qemu.go

```diff
@@ -83,67 +86,51 @@ func StartEVEQemu(qemuARCH, qemuOS, eveImageFile, qemuSMBIOSSerial string, eveTe
 	if qemuMonitorPort != 0 {
 		qemuOptions += fmt.Sprintf("-monitor tcp:localhost:%d,server,nowait  ", qemuMonitorPort)
 	}
-	nets, err := utils.GetSubnetsNotUsed(1)
-	if err != nil {
-		return fmt.Errorf("StartEVEQemu: %s", err)
-	}
-	offset := 0
-	network := nets[0].Subnet
-	var ethIndex int
-	qemuOptions += fmt.Sprintf("-netdev user,id=eth%d,net=%s,dhcpstart=%s,ipv6=off",
-		ethIndex, network, nets[0].FirstAddress)
-	for k, v := range qemuHostFwd {
-		origPort, err := strconv.Atoi(k)
-		if err != nil {
-			log.Errorf("Failed converting %s to Integer", k)
-			break
+
+	if withSDN {
+		// Ports connecting SDN VM with EVE VM.
+		socketPort := netDevBasePort
+		for i, port := range netModel.Ports {
+			qemuOptions += fmt.Sprintf("-netdev socket,id=eth%d,connect=:%d", i, socketPort)
+			qemuOptions += fmt.Sprintf(" -device %s,netdev=eth%d,mac=%s ", netDev, i,
+				port.EVEConnect.MAC)
+			socketPort++
 		}
-		newPort, err := strconv.Atoi(v)
+	} else {
+		// Use SLIRP networking to connect QEMU VM with the host.
+		nets, err := utils.GetSubnetsNotUsed(1)
 		if err != nil {
-			log.Errorf("Failed converting %s to Integer", v)
-			break
+			return fmt.Errorf("StartEVEQemu: %s", err)
 		}
-		qemuOptions += fmt.Sprintf(",hostfwd=tcp::%d-:%d", origPort+offset, newPort+offset)
-	}
-	qemuOptions += fmt.Sprintf(" -device %s,netdev=eth%d ", netDev, ethIndex)
-	offset += 10
-	ethIndex++
+		network := nets[0].Subnet
+		for i, port := range netModel.Ports {
+			qemuOptions += fmt.Sprintf("-netdev user,id=eth%d,net=%s,dhcpstart=%s,ipv6=off",
+				i, network, nets[0].FirstAddress)
```

Comment:
Current behavior is to use nets[0].SecondAddress for the second dhcpstart. Without this we may have the same addresses on EVE interfaces as I can see in my experiments. In that case I cannot connect to EVE and tests indicates that problem as well.

Reply:
Fixed

Reply:
Thank you!